src/aarch64/ffi.c

```diff
@@ -927,6 +930,10 @@ ffi_call_int (ffi_cif *cif, void (*fn)(void), void *orig_rvalue,
 #endif
     }
 
+#ifdef __CHERI_PURE_CAPABILITY__
+  context->x9 = (XREG)__builtin_cheri_bounds_set(stack, state.nsaa);
```

Comment:
This doesn't handle the case where it's not representable. In that instance, the stack region (a) needs to be overaligned (b) needs to be padded.